src/shared/format-table.c

```diff
@@ -1094,6 +1106,25 @@ static const char *table_data_format(TableData *d) {
                 break;
         }
 
+        case TABLE_IFINDEX: {
+                _cleanup_free_ char *p;
+                char name[IF_NAMESIZE + 1];
```

Comment:
This needs to be initialized, because `if_indextoname` doesn't do termination.
Also, there's no need to use `new` below, we have `asprintf` for that.

I'd prefer to have this split out to a new function `ifindex_to_name` or so, that'd simply return the formatted string. It could then be used to replace similar code in `machinectl.c` and possibly other places.

Oh, I see that the code in networkctl.c that calls `if_indextoname` is busted. Something to fix too :)